### PR TITLE
`CI`: added workaround for Snapshots in `Xcode Cloud`

### PR DESCRIPTION
**Pull request by NachoSoto**
See https://github.com/pointfreeco/swift-snapshot-testing/discussions/553.
Blocked by https://github.com/codecov/uploader/issues/1126.

### CODE REVIEW

File: Tests/TestPlans/XC-AllTests.xctestplan [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+  ],
+  "defaultOptions" : {
+    "codeCoverage" : false,
+    "environmentVariableEntries" : [
+      {
+        "key" : "RCRunningTests",
+        "value" : "1"
+      },
+      {
+        "key" : "XCODE_CLOUD",
```
Comment (NachoSoto) [inline Tests/TestPlans/XC-AllTests.xctestplan:19]: New test plan to have this.

File: Tests/UnitTests/Networking/Backend/__Snapshots__/BackendLoginTests/iOS17-testLoginWithFailedVerification.1.json [hunk 2]
```diff
@@ -0,0 +1,15 @@
+{
+  "headers" : {
```
Comment (NachoSoto) [inline Tests/UnitTests/Networking/Backend/__Snapshots__/BackendLoginTests/iOS17-testLoginWithFailedVerification.1.json:2]: Some of these were missing.

File: Tests/UnitTests/Networking/Backend/__Snapshots__/BackendPostReceiptDataTests/iOS17-testGetsEntitlementsWithVerifiedResponse.1.json [hunk 3]
```diff
@@ -1,6 +1,8 @@
 {
   "headers" : {
-    "Authorization" : "Bearer asharedsecret"
+    "Authorization" : "Bearer asharedsecret",
+    "X-Nonce" : "MTIzNDU2Nzg5MGFi",
+    "X-Post-Params-Hash" : "app_user_id,fetch_token:sha256:2cc3741ddb808ae6fd179f1398d9115b58772590765edf799ec778f929faa46d"
```
Comment (NachoSoto) [inline Tests/UnitTests/Networking/Backend/__Snapshots__/BackendPostReceiptDataTests/iOS17-testGetsEntitlementsWithVerifiedResponse.1.json:5]: Some of these were outdated.

File: Tests/UnitTests/Purchasing/ProductRequestDataTests.swift [hunk 4]
```diff
@@ -218,10 +217,10 @@ class ProductRequestDataTests: TestCase {
 
 private extension ProductRequestDataTests {
 
-    func assertSnapshot(_ data: ProductRequestData,
-                        testName: String = #function,
-                        line: UInt = #line) {
-        SnapshotTesting.assertSnapshot(
```
Comment (NachoSoto) [inline Tests/UnitTests/Purchasing/ProductRequestDataTests.swift:224]: This needed to call our overload now.

File: Tests/UnitTests/TestHelpers/SnapshotTesting+Extensions.swift [hunk 5]
```diff
@@ -13,9 +13,77 @@
...
+    let snapshotDirectory: String?
+
+    if ProcessInfo.isXcodeCloud {
+        let ciPathPrefix = "/Volumes/workspace/repository/ci_scripts/"
+        let components = URL(string: file.description)!.pathComponents
+        let projectIndex = components.firstIndex(of: "repository")!
+        let folders = components[(projectIndex + 1)..<components.endIndex - 1].joined(separator: "/")
+        let fileName = (components[components.endIndex - 1] as NSString).deletingPathExtension
+
+        snapshotDirectory = "\(ciPathPrefix)\(folders)/__Snapshots__/\(fileName)"
```
Comment (NachoSoto) [inline Tests/UnitTests/TestHelpers/SnapshotTesting+Extensions.swift:67]: Necessary for Xcode Cloud to be able to read snapshots.

File: ci_scripts/Tests [hunk 6]
```diff
@@ -0,0 +1 @@
+../Tests/
```
Comment (NachoSoto) [inline ci_scripts/Tests:1]: Necessary for Xcode Cloud to have access to test snapshots.

**Comment by NachoSoto** [comment]
This is ready.

**Review by tonidero — APPROVED** [review]

**Comment by NachoSoto** [comment]
I have no idea what's going on with Codecov :(

**Comment by tonidero** [comment]
Hmm tests seems to have passed, and codecov seems to have gotten stuck... Did you try rerunning tests? (seems like the last run was 6 days ago)

**Comment by NachoSoto** [comment]
Yeah I re-ran several times and tried increasing timeouts. No luck, so I filed a but with them: https://github.com/codecov/feedback/issues/50

**Pull request merged**